Cluster

afeefghannam89 opened the pull request:
This pull request will support the installation and configuration of x Redis instances on x nodes as a cluster or standalone.

afeefghannam89 committed:
Install redis

afeefghannam89 committed:
Manage THP and overcommit memory

afeefghannam89 committed:
* Install redis
* Configure redis instances
* Build redis cluster

afeefghannam89 committed:
* Enhance cluster build
* Add some changes to configurations

afeefghannam89 committed:
Add docs

afeefghannam89 committed:
Change some variables and add tags

afeefghannam89 committed:
Change variable defaults. Fix some conditions

afeefghannam89 committed:
Add Pipeline

afeefghannam89 committed:
Add workflow

afeefghannam89 committed:
Change action run to all branches

afeefghannam89 committed:
Change action run to all branches

afeefghannam89 committed:
Add requirements

afeefghannam89 committed:
Fix scenario

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Molecule fix

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Remove extra space

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Add a new pipeline

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Remove centos7

afeefghannam89 committed:
Modify verify.yml

afeefghannam89 committed:
Modify verify.yml

afeefghannam89 committed:
Modify verify.yml and converge

afeefghannam89 committed:
Modify verify.yml and converge

afeefghannam89 committed:
Modify verify.yml and converge

afeefghannam89 committed:
Modify verify.yml and converge

afeefghannam89 committed:
Molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix condition in template

afeefghannam89 committed:
Fix verify.yml error

afeefghannam89 committed:
Add new pipeline

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
molecule

afeefghannam89 committed:
Fix molecule and enhance the cluster command

afeefghannam89 committed:
fix variable fehler

afeefghannam89 committed:
fix variable fehler

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Add extra nodes

afeefghannam89 committed:
Add extra nodes

afeefghannam89 committed:
fix molecule

afeefghannam89 committed:
fix molecule

afeefghannam89 committed:
Add a new pipeline

afeefghannam89 committed:
Fix molecle

afeefghannam89 committed:
Verify redis standalone without TLS

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
test molecule

afeefghannam89 committed:
Add verify to standalone with tls

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Fix molecule

afeefghannam89 committed:
Add verfy for cluster without TLS

afeefghannam89 committed:
Add verify for cluster with TLS

afeefghannam89 committed:
Fix mlecule

afeefghannam89 committed:
Fix mlecule

afeefghannam89 committed:
Test

afeefghannam89 committed:
* Update docs
* Changes to template and configurations

afeefghannam89 committed:
Enable pipeline run on all branches

afeefghannam89 committed:
Add a variable

afeefghannam89 committed:
Update docs and a varibale

afeefghannam89 committed:
Add molecule test results to docs

afeefghannam89 committed:
Update docs

widhalmt commented on .ansible-lint:
```diff
@@ -0,0 +1,6 @@
+skip_list:
+  - yaml
+  - fqcn-builtins  # Use FQCN for builtin actions.
```
That's a nice compromise. Use fqcn in collections but don't overdo it with builtins. I like that!

widhalmt commented on README.md:
```diff
@@ -1 +1,81 @@
+
+The current version is tested on rocky 8.
+
+You can use the [ca role](https://github.com/NETWAYS/ansible-role-ca), if you want to use self singed certificate.
```
```suggestion
You can use the [ca role](https://github.com/NETWAYS/ansible-role-ca), if you want to create a local CA and use its certificates.
```

widhalmt commented on README.md:
```diff
@@ -1 +1,81 @@
+
+## Variables ##
+
+* `manage_dnf_module`: Should the corresponding dnf module enabled? (default: `true`)
```
Is this a redis specific variable? If it's not to be used in more than one role, please stick to the naming scheme and prefix it wit the name of the role.

widhalmt commented on README.md:
```diff
@@ -1 +1,81 @@
+
+* `manage_dnf_module`: Should the corresponding dnf module enabled? (default: `true`)
+* `redis_version`: Redis version to be installed (default: `6`)
+* `kernel_virtual_memory`: If you want to disable the kernel virtual memory, set the value to 1 (default: `undefined, the system default will be used`)
```
Is this a redis specific variable? If it's not to be used in more than one role, please stick to the naming scheme and prefix it wit the name of the role.

widhalmt commented on README.md:
```diff
@@ -1 +1,81 @@
+* `manage_dnf_module`: Should the corresponding dnf module enabled? (default: `true`)
+* `redis_version`: Redis version to be installed (default: `6`)
+* `kernel_virtual_memory`: If you want to disable the kernel virtual memory, set the value to 1 (default: `undefined, the system default will be used`)
+* `transparent_hugepages_value`: Manage THP, availible values are `always`, `madvise` or `never` (default: `undefined, the system default will be used`)
```
Is this a redis specific variable? If it's not to be used in more than one role, please stick to the naming scheme and prefix it wit the name of the role.

widhalmt commented on README.md:
````diff
@@ -1 +1,81 @@
+    - repl-diskless-sync-delay 5
+    - repl-diskless-load disabled
+    ```
+## Example Playbook ##
+
+```yaml
+---
+- hosts: redis
+  become: yes
+  roles:
+    - redis_cluster
+```
````
Will this role run without any variables set? If yes, then: greta. If not, please add the minimum of variables to the playbook.

widhalmt replied:
As seen in https://github.com/NETWAYS/ansible-role-redis-cluster/pull/2/files#diff-b1d26d8e66ae986086a4d667f0b0bab5f3365cfe0a394be4ae68bb1d5ed200b1R9-R10 , the role runs without any mandatory variables being set. So yes, that's great.

widhalmt commented on tasks/cluster.yml:
```diff
@@ -0,0 +1,47 @@
+    var: cluster_info
+  tags: redis_cluster
+
+- name: Creat a redis cluster
```
```suggestion
- name: Create a redis cluster
```
Typo. :-)

widhalmt commented on tasks/cluster.yml:
```diff
@@ -0,0 +1,47 @@
+          {{ redis_instances.append( host_ip+ ':' +host_port ) }}
+        {%- endfor -%}
+      {%- endfor -%}
+      redis-cli --cluster create {{ redis_instances | join(' ') }} --cluster-replicas {{ redis_masters_replicas }} -a {{ redis_requirepass }} --cluster-yes {% if redis_configurations[0]['redis_tls_cluster'] is defined and redis_configurations[0]['redis_tls_cluster'] == 'yes' %} --tls --cert {{ redis_configurations [0]['redis_tls_cert_file'] }} --key {{ redis_configurations [0]['redis_tls_key_file ...
```
Since you're already using YAML multiline, you could put every option in a separate line. This would help with changing the command later if there are any new changes to be done.
```suggestion
      redis-cli
      --cluster
      create {{ redis_instances | join(' ') }}
      --cluster-replicas {{ redis_masters_replicas }}
      ...
```
(I'm not writing the whole command, because 1. I'm lazy and 2. I don't want to hijack your line and mark it as my own. :-) )

widhalmt reviewed (changes requested):
WOW, that's a mighty big change. I'm absolutely impressed.

The changes I put in are all very small. Please feel free to close conversations when there's nothing more to say. Like when I left a comment or so.

widhalmt committed:
Update README.md

Clarify what certificates get created

widhalmt committed:
Add workaround for cgroups

Refer to
https://github.com/geerlingguy/docker-rockylinux8-ansible/pull/3#issuecomment-1282963368
for details

widhalmt committed:
Update tasks/cluster.yml

Typo

widhalmt committed:
Rename variables to match nomenclature

afeefghannam89 committed:
Linebreak long task

widhalmt committed:
Merge branch 'cluster' of github.com:NETWAYS/ansible-role-redis-cluster into cluster

widhalmt committed:
Mixed up variable names

widhalmt reviewed (approved):
I put in a few changes. I guess, that was the last drop that was missing. Really great work, thank you.
